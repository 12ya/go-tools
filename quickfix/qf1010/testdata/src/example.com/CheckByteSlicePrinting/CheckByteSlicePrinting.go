package pkg

import "fmt"

type Stringable []byte

func (*Stringable) String() string { return "" }

func fn() {
	type ByteSlice []byte
	type Alias1 = []byte
	type Alias2 = ByteSlice
	type Alias3 = byte
	type Alias4 = []Alias3
	var b1 []byte
	var b2 ByteSlice
	var b3 *Stringable
	var b4 Stringable

	var s string
	fmt.Print(1, b1, 2, []byte(""), b2, s)       //@ diag(`could convert argument to string`), diag(`could convert argument to string`), diag(`could convert argument to string`)
	fmt.Print(Alias1{})                          //@ diag(`could convert argument to string`)
	fmt.Print(Alias2{})                          //@ diag(`could convert argument to string`)
	fmt.Print(Alias4{})                          //@ diag(`could convert argument to string`)
	fmt.Fprint(nil, 1, b1, 2, []byte(""), b2, s) //@ diag(`could convert argument to string`), diag(`could convert argument to string`), diag(`could convert argument to string`)
	fmt.Print()
	fmt.Fprint(nil)

	fmt.Println(b3)
	fmt.Println(b4) //@ diag(`could convert argument to string`)
}
