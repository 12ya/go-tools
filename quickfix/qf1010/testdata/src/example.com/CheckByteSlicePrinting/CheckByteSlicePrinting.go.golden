package pkg

import "fmt"

type Stringable []byte

func (*Stringable) String() string { return "" }

func fn() {
	type ByteSlice []byte
	type Alias1 = []byte
	type Alias2 = ByteSlice
	type Alias3 = byte
	type Alias4 = []Alias3
	var b1 []byte
	var b2 ByteSlice
	var b3 *Stringable
	var b4 Stringable

	var s string
	fmt.Print(1, string(b1), 2, string([]byte("")), string(b2), s)       //@ diag(`could convert argument to string`), diag(`could convert argument to string`), diag(`could convert argument to string`)
	fmt.Print(string(Alias1{}))                                          //@ diag(`could convert argument to string`)
	fmt.Print(string(Alias2{}))                                          //@ diag(`could convert argument to string`)
	fmt.Print(string(Alias4{}))                                          //@ diag(`could convert argument to string`)
	fmt.Fprint(nil, 1, string(b1), 2, string([]byte("")), string(b2), s) //@ diag(`could convert argument to string`), diag(`could convert argument to string`), diag(`could convert argument to string`)
	fmt.Print()
	fmt.Fprint(nil)

	fmt.Println(b3)
	fmt.Println(string(b4)) //@ diag(`could convert argument to string`)
}
